railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
+threads threads_count, threads_count
```

Comment:
You have to set a minimum. Setting minim == maximum means that 5 threads get spawned at start time instead of waiting to spawn threads at request time. The alternative would be to set to 1.